dspace-jspui/src/main/java/org/dspace/app/webui/servlet/BitstreamServlet.java

```diff
@@ -182,27 +198,78 @@ protected void doDSGet(Context context, HttpServletRequest request,
         
         // Modification date
         // Only use last-modified if this is an anonymous access
+        // Set Cache Control headers appropriately.
         // - caching content that may be generated under authorisation
         //   is a security problem
+        // Gather last modified timestamp if possible
+        Date lastModified;
+
+        if(item != null)
+            lastModified = item.getLastModified();
+        else
+            lastModified = null;
+
+
+        boolean isAnonymouslyReadable = false;
+
         if (context.getCurrentUser() == null)
         {
-            // TODO: Currently the date of the item, since we don't have dates
-            // for files
-            response.setDateHeader("Last-Modified", item.getLastModified()
-                    .getTime());
+            isAnonymouslyReadable = true;
```

Comment:
I don't understand why the session being anonymous means that the bitstream is anonymously readable.